nav2_behavior_tree/CMakeLists.txt

```diff
@@ -211,23 +203,43 @@ add_library(nav2_goal_updated_controller_bt_node SHARED plugins/decorator/goal_u
 list(APPEND plugin_libs nav2_goal_updated_controller_bt_node)
 
 foreach(bt_plugin ${plugin_libs})
-  ament_target_dependencies(${bt_plugin} ${dependencies})
+  target_include_directories(${bt_plugin}
+    PRIVATE
+    "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+    "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>")
+  target_link_libraries(${bt_plugin} PUBLIC
```

Comment:
Question: Is there much of a speed up by specifying the exact targets vs all the targets in the packages? It is somewhat convenient to be able to specify the `dependencies` variable and just target the set as a whole without having to think about it much. But if this is a big speed up, that is a worthwhile trade off to me to specify them individually.

Reply:
So yeah, you get most of the speed-up by specifying the exact targets, and splitting this into PUBLIC and PRIVATE.  That is, anything that is marked as PRIVATE will *not* be exposed to the downstream targets, thus shortening the list of things they have to link against later on.

Unfortunately, as it stands, `ament_target_dependencies` does not fully support this (in particular, it doesn't support PRIVATE).  We *could* add it to `ament_target_dependencies`, but since we are trying to slowly move away from it, I didn't think it was particularly worth it.

While thinking about it, I did come up with another way to do this, which would reduce duplication.  Inside of `nav2_common`, we could have a "map" between a package name and target names.  And then we could have a custom `cmake` macro/function here that, given a target and a list of packages, automatically looked up the mapping between a package name and a target name, and called `target_link_libraries`.  The caller would call something like `nav2_link_libraries(TARGET mytarget PACKAGES foo bar)`.  The downside is that it is more CMake "magic", which is is something that people have complained about for years with `ament`.  I also don't know what the impact on compile times will be.  Nevertheless, if you think this is interesting, I can look into prototyping it.